service/policy/db/query.sql

```diff
@@ -230,6 +235,121 @@ WHERE
 GROUP BY
     ad.id, an.id, nfq.fqn, n_grants.grants;
 
+-- name: GetAttribute :one
+SELECT
+    ad.id,
+    ad.name as attribute_name,
+    ad.rule,
+    JSON_STRIP_NULLS(JSON_BUILD_OBJECT('labels', ad.metadata -> 'labels', 'created_at', ad.created_at, 'updated_at', ad.updated_at)) AS metadata,
+    ad.namespace_id,
+    ad.active,
+    n.name as namespace_name,
+    ad.values_order,
+    JSON_AGG(JSON_BUILD_OBJECT('id', avt.id, 'value', avt.value, 'active', avt.active)) AS values,
+    JSONB_AGG(
+        DISTINCT JSONB_BUILD_OBJECT(
+            'id', kas.id,
+            'uri', kas.uri,
+            'public_key', kas.public_key
+        )
+    ) FILTER (WHERE adkag.attribute_definition_id IS NOT NULL) AS grants,
+    fqns.fqn
+FROM attribute_definitions ad
+LEFT JOIN attribute_namespaces n ON n.id = ad.namespace_id
+LEFT JOIN (
+    SELECT
+        av.id,
+        av.value,
+        av.active,
+        JSON_AGG(DISTINCT JSONB_BUILD_OBJECT('id', vkas.id,'uri', vkas.uri,'public_key', vkas.public_key )) FILTER (WHERE vkas.id IS NOT NULL AND vkas.uri IS NOT NULL AND vkas.public_key IS NOT NULL) AS val_grants_arr,
+        av.attribute_definition_id
+    FROM attribute_values av
+    LEFT JOIN attribute_value_key_access_grants avg ON av.id = avg.attribute_value_id
+    LEFT JOIN key_access_servers vkas ON avg.key_access_server_id = vkas.id
+    GROUP BY av.id
+) avt ON avt.attribute_definition_id = ad.id
+LEFT JOIN attribute_definition_key_access_grants adkag ON adkag.attribute_definition_id = ad.id
+LEFT JOIN key_access_servers kas ON kas.id = adkag.key_access_server_id
+LEFT JOIN attribute_fqns fqns ON fqns.attribute_id = ad.id AND fqns.value_id IS NULL
+WHERE ad.id = $1
+GROUP BY ad.id, n.name, fqns.fqn;
+
+-- name: CreateAttribute :one
+INSERT INTO attribute_definitions (namespace_id, name, rule, metadata)
+VALUES (@namespace_id, LOWER(@name), @rule, @metadata)
+RETURNING id, name,
+    JSON_STRIP_NULLS(JSON_BUILD_OBJECT('labels', metadata -> 'labels', 'created_at', created_at, 'updated_at', updated_at)) as metadata;
+
+-- name: UpdateAttribute :one
+UPDATE attribute_definitions
+SET
+    name = COALESCE(LOWER(sqlc.narg('name')), name),
+    rule = COALESCE(sqlc.narg('rule'), rule),
+    values_order = COALESCE(sqlc.narg('values_order'), values_order),
+    metadata = COALESCE(sqlc.narg('metadata'), metadata),
+    active = COALESCE(sqlc.narg('active'), active)
+WHERE id = $1
+RETURNING id;
+
+-- name: DeleteAttribute :execrows
+DELETE FROM attribute_definitions WHERE id = $1;
+
+-- name: AssignKeyAccessServerToAttribute :execrows
+INSERT INTO attribute_definition_key_access_grants (attribute_definition_id, key_access_server_id)
+VALUES ($1, $2);
+
+-- name: RemoveKeyAccessServerFromAttribute :execrows
+DELETE FROM attribute_definition_key_access_grants
+WHERE attribute_definition_id = $1 AND key_access_server_id = $2;
+
+-- name: ListKeyAccessServerGrants :many
```

Comment:
was this statement changed, or was it a merge conflict issue perhaps? https://github.com/opentdf/platform/blob/32c09c3993bee369401537a9370478dc519ec415/service/policy/db/query.sql#L36

Reply:
Merge conflicts are fixed and all I did was move this query down with the other KAS ones